test/e2e/framework.go

```diff
@@ -419,3 +420,166 @@ func kubectlExec(namespace string, podName, containerName string, args ...string
 func KubeDescribe(text string, body func()) bool {
 	return Describe("[k8s.io] "+text, body)
 }
+
+// PodStateVerification represents a verification of pod state.
+// Any time you have a set of pods that you want to operate against or query,
+// this struct can be used to declaratively identify those pods.
+type PodStateVerification struct {
+	// Optional: only pods that have k=v labels will pass this filter.
+	selectors map[string]string
+
+	// Required: The phases which are valid for your pod.
+	validPhases []api.PodPhase
+
+	// Optional: only pods passing this function will pass the filter
+	// Verify a pod.
+	// As an optimization, in addition to specfying filter (boolean),
+	// this function allows specifying an error as well.
+	// The error indicates that the polling of the pod spectrum should stop.
+	verify func(api.Pod) (bool, error)
+
+	// Optional: only pods with this name will pass the filter.
+	podName string
+}
+
+type ClusterVerification struct {
+	client    *client.Client
+	namespace string
+	podState  PodStateVerification
+}
+
+func (f *Framework) NewClusterVerification(filter PodStateVerification) *ClusterVerification {
+	return &ClusterVerification{
+		f.Client,
+		f.Namespace.Name,
+		filter,
+	}
+}
+
+func passesPodNameFilter(pod api.Pod, name string) bool {
```

Comment:
I don't know how I feel about the name "passes", but I like the separation.  b/c what you are doing really is checking to see if objA ~= objB.  It's almost like string matching comparison.  

(match/compare/someothername)PodName(a, b)